servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-safe-by-default.adoc

```diff
@@ -42,121 +43,27 @@ The implication of the above approach is the following:
-    Publisher.from(1, 2, 3, 4)
-            .afterOnNext(integer -> {
-                latch.countDown();  # <2>
-            })
-            .beforeRequest(requestN -> {
-                latch.await(); # <3>
-            });
-----
-<1> Hypothetical synchronization point. In real life it may be due to the code waiting for an event to happen externally.
-<2> Trigger the external event (hypothetical synchronization point of `CountDownLatch`) **after** receiving the item.
-<3> Wait for the external event to happen (hypothetical synchronization point of `CountDownLatch`) **before** sending
-`requestN` to the `Publisher`.
-
-As per ReactiveStreams link:https://github.com/reactive-streams/reactive-streams-jvm#1.1[rule 1.1], request for items
-**MUST** happen before the items are delivered. In the above code, we are waiting for an item to be emitted before
-sending a request to the `Publisher`. This results in a deadlock as an item can not be emitted by the source without a
-request being received and user code making sure that the request is not sent before an item is emitted.
-
-In order to avoid such scenarios, it is handy to follow certain best practices while writing blocking code in operators:
-
-- Avoid co-ordination between two operators on the same source.
-- If such co-ordination is required, try limiting such coordination in either data or control path but not
-inter-dependent on each other.
-- If co-ordination is required between data and control path, be aware of ReactiveStreams semantics and how the two
-paths interact with each other.
-
-If these rules are followed the above example can be modified to:
-
-[source, java]
-----
-    CountDownLatch latch = new CountDownLatch(1);
-    Publisher.from(1, 2, 3, 4)
-            .afterOnNext(integer -> {
-                latch.countDown();
-            })
-            .afterRequest(requestN -> { # <1>
-                latch.await();
-            });
-----
-<1> Use `afterRequest` which happens **after** `requestN` is delivered to the source.
-
-In this modified example, since we now use `afterRequest`, instead of `beforeRequest`, we do not block `requestN` to
-go to the source and this code is safe.
+== Task Offloading
+ServiceTalk uses Netty for network I/O and the Netty implementation uses a fixed number of I/O threads for executing
+network operations. To maximize throughput and minimize latency it is important to ensure that the I/O threads are
+always available to respond immediately to network activity. One approach to ensure I/O thread availability is to
+carefully limit the scope of work done by I/O threads and, whenever practical, delegate all other necessary tasks that
+are not related to I/O to some other thread. Moving tasks from I/O threads to other threads is called “offloading” and
+is a core technique used by ServiceTalk.
+
+ServiceTalk will, by default, execute most application code on threads other than the Netty I/O threads. There are some
```

Comment:
`There are some...` -> this seems like a caveat that I'm not sure make sense calling out in high level docs. In these cases it is unlikely people will block and unlikely to cause an issue, can we move/remove this cognitive overhead?